### PR TITLE
[CWS] do not fork from the wrong parent

### PR DESCRIPTION
**Pull request by safchain**
### What does this PR do?

This fixes:

```
agent -> [fork] -> agent -> [exec loss] -> dash -> [fork] -> dash
```
generating
```
agent -> [fork] -> agent -> [fork] -> agent
```

The node is inserted but without a parent. This will generate a broken lineage error.

### Reviewer's Checklist

- [ ] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [ ] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [ ] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [ ] At least one `team/..` label has been applied, indicating the team(s) that should QA this change.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the `k8s/<min-version>` label, indicating the lowest Kubernetes version compatible with this feature.
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

**Review by spikat — APPROVED** [review]

File: pkg/security/probe/probe_linux.go [hunk 1]
```diff
@@ -484,6 +484,38 @@ func (p *Probe) onEventLost(perfMapName string, perEvent map[string]uint64) {
...
+// setProcessContext set the process context, should return false if the event shouldn't be dispatched
+func (p *Probe) setProcessContext(eventType model.EventType, event *model.Event) bool {
+	entry, isResolved := p.fieldHandlers.ResolveProcessCacheEntry(event)
+	if !eventWithNoProcessContext(eventType) {
+		if !isResolved {
+			event.Error = &ErrNoProcessContext{Err: errors.New("process context not resolved")}
+		} else if !entry.HasValidLineage() {
+			event.Error = &ErrProcessBrokenLineage{PIDContext: entry.PIDContext}
+			p.resolvers.ProcessResolver.CountBrokenLineage()
+		}
```
Comment (spikat) [inline pkg/security/probe/probe_linux.go:496]: Should we return false here to avoid dispatching a broken event?

Comment (safchain) [reply pkg/security/probe/probe_linux.go:496]: We reject the broken event at the CWS consumer side as we don't want to reject the event for the other consumers that may not be impacted by broken lineage. https://github.com/DataDog/datadog-agent/blob/main/pkg/security/rules/engine.go#L425

Comment (spikat) [reply pkg/security/probe/probe_linux.go:496]: I see, thanks ! :)

File: pkg/security/secl/model/model_unix.go [hunk 2]
```diff
@@ -337,8 +337,10 @@ type Process struct {
 	ScrubbedArgsTruncated bool           `field:"-" json:"-"`
 	Variables             eval.Variables `field:"-" json:"-"`
 
-	IsThread    bool `field:"is_thread"` // SECLDoc[is_thread] Definition:`Indicates whether the process is considered a thread (that is, a child process that hasn't executed another program)`
-	IsExecChild bool `field:"-"`         // Indicates whether the process is an exec child of its parent
+	IsThread        bool `field:"is_thread"` // SECLDoc[is_thread] Definition:`Indicates whether the process is considered a thread (that is, a child process that hasn't executed another program)`
+	IsExecChild     bool `field:"-"`         // Indicates whether the process is an exec child of its parent
+	IsParentMissing bool `field:"-"`         // Indicates the direct parent is missing
+	IsNotAccurate   bool `field:"-"`         // Indicates the entry may be not accurate
```
Comment (Gui774ume) [inline pkg/security/secl/model/model_unix.go:343]: Maybe add "because it comes from the /proc filesystem" ?

**Review by Gui774ume — APPROVED** [review]

**Pull request merged**